gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/langchain_factory.py

```diff
@@ -142,3 +153,40 @@ def get_vector_store_factory(
         )
     else:
         raise VectorStoreUnknownException()
+
+
+def get_callback_handler_factory(setting: BaseObservabilitySetting) -> LangChainCallbackHandlerFactory:
+    """
+    Creates a Langchain Callback Handler Factory according to the given setting
+    Args:
+        setting: The Observability setting
+
+    Returns:
+        The Observability Factory, or raise an exception otherwise
+    """
+
+    logger.info('Get Observability Factory for the given setting')
+    if isinstance(setting, LangfuseObservabilitySetting):
+        logger.debug('Observability Factory - OpenAIObservabilityFactory')
+        return LangfuseCallbackHandlerFactory(setting=setting)
+    else:
+        raise GenAIUnknownObservabilityProviderSettingException()
+
+
+def create_langfuse_callback_handler(
```

Comment:
Rename it to be generic, langfuse is a specific observability provider name.

```suggestion
def create_observability_callback_handler(
```

Take care to correctly refactor it I may have miss some occurrences.